src/backend/cdb/cdbmutate.c

```diff
@@ -993,6 +998,7 @@ shareinput_mutator_xslice_2(Node *node, PlannerInfo *root, bool fPop)
 
 		sisc->producer_slice_id = pershare->producer_slice_id;
 		sisc->nconsumers = bms_num_members(pershare->participant_slices) - 1;
+		currentSlice->directDispatch = pershare->directDispatch;
```

Comment:
You care about copying content lists, but assign the same DDI for several slices. Why are both actions correct?

Reply:
Assigning the same DDI is acceptable because this is the last time it's modified - after this point it will only be read. However, it seems like copying isn't necessary anymore either, because we are simply merging all of them, updating many of them at once shouldn't be an issue. I will remove the unnecessary copies